lang/python/python-augeas/patches/001-fix-ffi.patch

```diff
@@ -0,0 +1,14 @@
+--- a/augeas/ffi.py
++++ b/augeas/ffi.py
+@@ -43,8 +43,10 @@ const char *aug_error_details(augeas *au
+ 
+ void free(void *);
+ """)
++import os
+ 
+-lib = ffi.dlopen("augeas")
++if os.path.isfile('/etc/openwrt_version'):
++	lib = ffi.dlopen("libaugeas.so")
```

Comment:
This code is executed during build (to build the cffi module), not during runtime. (This is why host cffi is necessary.)

I think what you want to do (though I haven't tested this) is to point it to the augeas shared library in `$(STAGING_DIR)` (installed during Build/InstallDev), but this may not work because the shared library is built for a different architecture than the build host.

Reply:
Sorry, I think my previous comment is wrong, at least partially (the code is called during runtime, but I think it is also being called during build).

Reply:
Without ffi patch build was failing on hosts without augeas (it also failed here in circleci)

Reply:
I imagine looking for `libaugeas.so` instead of `augeas` is probably necessary. Testing for `/etc/openwrt_version` should be unnecessary because this file (ffi.py) should be loaded either during build or during run-time, but not both.

I just opened hercules-team/python-augeas#48 - I suggest waiting a few days to see how that issue is decided, which will affect how the fix is coded and thus affect this patch.

Reply:
Thanks !